nimare/meta/utils.py

```diff
@@ -359,19 +363,33 @@ def compute_kda_ma(
         idx = (np.min(sphere, 1) >= 0) & (np.max(np.subtract(sphere, shape), 1) <= -1)
         sphere = sphere[idx, :].astype(int)
         exp = exp_idx[i]
-        if sum_overlap:
-            kernel_data[exp][tuple(sphere.T)] += value
+        if memmap_filename:
+            if sum_overlap:
+                kernel_data[exp][tuple(sphere.T)] += value
+            else:
+                kernel_data[exp][tuple(sphere.T)] = value
+            if i % chunk_size == 0:
+                kernel_data.flush()
         else:
-            kernel_data[exp][tuple(sphere.T)] = value
-
-        if memmap_filename and i % chunk_size == 0:
-            # Write changes to disk
-            kernel_data.flush()
+            if kernel_data_lst[exp] is None:
+                kernel_data = np.zeros(shape, dtype=type(value))
```

Comment:
I don't think we want to create a zeros array, as this will take up the memory footprint of `shape` in memory, even if its all zeros.

Reply:
Agreed. It is only 7MB for the peak memory, but it will be better to start using sparse arrays from coordinates directly.

Reply:
Ah okay, it's probably optimized to not take up as much memory as it could when empty